lists/list_cache.go

```diff
@@ -147,6 +147,8 @@ func logger() *logrus.Entry {
 }
 
 // downloads and reads files with domain names and creates cache for them
+//
+//nolint:funlen // will refactor in a later commit
```

Comment:
I'll refactor this in the follow-up.  
I already have something locally, but think it'll be better in the PR that allows the `HostsFileResolver` to use multiple sources since that is going to look a lot like this function.